chips/x86_q35/src/chip.rs

```diff
@@ -159,60 +185,39 @@ impl<'a, const PR: u16> Chip for Pc<'a, PR> {
     }
 }
 
-/// Component helper for constructing a [`Pc`] chip instance.
-///
-/// During the call to `finalize()`, this helper will perform low-level initialization of the PC
-/// hardware to ensure a consistent CPU state. This includes initializing memory segmentation and
-/// interrupt handling. See [`x86::init`] for further details.
-pub struct PcComponent<'a> {
-    pd: &'a mut PD,
-    pt: &'a mut PT,
+/// Default x86 PC peripherals commonly present on Q35 machines.
+pub struct PcDefaultPeripherals<const PR: u16 = RELOAD_1KHZ> {
+    pub com1: &'static SerialPort<'static>,
+    pub com2: &'static SerialPort<'static>,
+    pub com3: &'static SerialPort<'static>,
+    pub com4: &'static SerialPort<'static>,
+    pub pit: Pit<'static, PR>,
+    pub vga: &'static VgaText<'static>,
 }
 
-impl<'a> PcComponent<'a> {
-    /// Creates a new `PcComponent` instance.
+impl<const PR: u16> PcDefaultPeripherals<PR> {
+    /// Create and initialize default peripherals.
     ///
-    /// ## Safety
-    ///
-    /// It is unsafe to construct more than a single `PcComponent` during the entire lifetime of the
-    /// kernel.
-    ///
-    /// Before calling, memory must be identity-mapped. Otherwise, introduction of flat segmentation
-    /// will cause the kernel's code/data to move unexpectedly.
+    /// The caller must provide statics through `x86_q35_peripherals_static!()`.
     ///
-    /// See [`x86::init`] for further details.
-    pub unsafe fn new(pd: &'a mut PD, pt: &'a mut PT) -> Self {
-        Self { pd, pt }
-    }
-}
-
-impl Component for PcComponent<'static> {
-    type StaticInput = (
-        <SerialPortComponent as Component>::StaticInput,
-        <SerialPortComponent as Component>::StaticInput,
-        <SerialPortComponent as Component>::StaticInput,
-        <SerialPortComponent as Component>::StaticInput,
-        &'static mut MaybeUninit<Pc<'static>>,
-    );
-    type Output = &'static Pc<'static>;
-
-    fn finalize(self, s: Self::StaticInput) -> Self::Output {
-        // Low-level hardware initialization. We do this first to guarantee the CPU is in a
-        // predictable state before initializing the chip object.
+    /// ## Safety
+    /// - Must be called only once per kernel lifetime.
+    pub unsafe fn new(
+        s: (
+            (&'static mut core::mem::MaybeUninit<SerialPort<'static>>,),
+            (&'static mut core::mem::MaybeUninit<SerialPort<'static>>,),
+            (&'static mut core::mem::MaybeUninit<SerialPort<'static>>,),
+            (&'static mut core::mem::MaybeUninit<SerialPort<'static>>,),
+            &'static mut core::mem::MaybeUninit<VgaText<'static>>,
+        ),
+        page_dir: &mut PD,
+    ) -> Self {
+        // CPU/interrupt controller/VGA baseline init
         unsafe {
             x86::init();
             crate::pic::init();
```

Comment:
For sam4l etc. we do this in main.rs immediately before the call to create the peripherals; but I think doing it in here is effectively no different